domain/miningmanager/mempool/transactions_pool.go

```diff
@@ -204,17 +209,52 @@ func (tp *transactionsPool) limitTransactionCount() error {
 	return nil
 }
 
-func (tp *transactionsPool) getTransaction(transactionID *externalapi.DomainTransactionID) (*externalapi.DomainTransaction, bool) {
+func (tp *transactionsPool) getTransaction(transactionID *externalapi.DomainTransactionID, clone bool) (*externalapi.DomainTransaction, bool) {
 	if mempoolTransaction, ok := tp.allTransactions[*transactionID]; ok {
+		if clone {
+			return mempoolTransaction.Transaction().Clone(), true
+		}
 		return mempoolTransaction.Transaction(), true
 	}
 	return nil, false
 }
 
-func (tp *transactionsPool) getAllTransactions() []*externalapi.DomainTransaction {
-	allTransactions := make([]*externalapi.DomainTransaction, 0, len(tp.allTransactions))
+func (tp *transactionsPool) getTransactionsByAddresses(clone bool) (
+	sending model.ScriptPublicKeyStringToDomainTransaction,
+	receiving model.ScriptPublicKeyStringToDomainTransaction,
+	err error) {
+	sending = make(model.ScriptPublicKeyStringToDomainTransaction)
+	receiving = make(model.ScriptPublicKeyStringToDomainTransaction)
+	var transaction *externalapi.DomainTransaction
 	for _, mempoolTransaction := range tp.allTransactions {
-		allTransactions = append(allTransactions, mempoolTransaction.Transaction())
+		if clone {
+			transaction = mempoolTransaction.Transaction().Clone()
+		} else {
+			transaction = mempoolTransaction.Transaction()
+		}
+		for _, input := range transaction.Inputs {
+			if input.UTXOEntry == nil { //this should be fixed
```

Comment:
Does it still happen?

Reply:
no, haven't encountered it since cloning.

Reply:
So let's remove the "this should be fixed" etc